openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -655,14 +660,20 @@ def _validate_password(password, username=None):
         errors.AccountPasswordInvalid
 
     """
+    if username == None or username == '':
+        username = DUMMY_USERNAME
```

Comment:
This approach seems surprising to me. Is it not better to keep username as a None value that we treat appropriately?

Reply:
Switch to if username is not defined, rather than defining a DUMMY_USERNAME, set the temp_user to None